src/services/services.ts

```diff
@@ -3961,18 +3975,15 @@ module ts {
             if (results) {
                 let sourceFile = getCanonicalFileName(normalizeSlashes(fileName));
 
-                // ensure the results are in the file we're interested in
-                results.forEach((value) => {
-                    let targetFile = getCanonicalFileName(normalizeSlashes(value.fileName));
-                    Debug.assert(sourceFile == targetFile, `Unexpected file in results. Found results in ${targetFile} expected only results in ${sourceFile}.`);
-                });
+                // Get occurrences only supports reporting occurrences for the file queried.  So 
+                // filter down to that list.
+                results = filter(results, r => r.fileName === fileName);
             }
 
             return results;
         }
 
-        /// References and Occurrences
-        function getOccurrencesAtPositionCore(fileName: string, position: number): ReferenceEntry[] {
+        function getDocumentHighlights(fileName: string, position: number): DocumentHighlights[] {

```

Comment:
Great - yeah I suspected as much with the diff. I'm still looking through to fix anything while we're already here.